### PR TITLE
incorporated changes from original corrupt branch

### PR DESCRIPTION
**Pull request by gtfisher**
Signed-off-by: gtfisher <grant@tarrant-fisher.co.uk>

### CODE REVIEW

File: chaosinstana/api.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        r = requests.get(url=url,
+                         params=params,
+                         headers={auth_header_key: auth_header_contents})
+
+        data = r.json()
+
+        return data
+    else:
+        logger.warning("No Instana Host or API Token Secrete were found.")
```
Comment (Lawouach) [inline chaosinstana/api.py]: I would reverse the if/else here:

```
if not (instana_host and instana_api_token):
   raise ActivityFailed("No Instana Host or API Token Secrete were found.")

url = "{}/api/events".format(
            configuration.get("instana_host"))

params = {}
if from_time:
   params["from"] = from_time

if to_time:
   params["to"] = to_time

 r = requests.get(url=url,
                         params=params,
                         headers={
                                 "Authorization": "apiToken {}".format(secrets.get("instana_api_token"))
    }
)

return r.json()
```

File: chaosinstana/probes.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+__all__ = ["get_all_events_in_window", "convert_time"]
+
+
+def convert_time(source_time: str) -> int:
+    """
+    If the `source_time` is an int, just use it as is, otherwise parse the
+    string using dateparser.parse, this should help with datestrings in
+    other langauges and also supports relative dates
+    for further info see https://pypi.org/project/dateparser/
+    """
```
Comment (Lawouach) [inline chaosinstana/probes.py:38]: Add these two lines:
```
if not source_time:
   return
```

File: chaosinstana/probes.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+def get_all_events_in_window(from_time: str = None, to_time: str = None,
+                             configuration: Configuration = None,
+                             secrets: Secrets = None) -> str:
+    """
+    get all events from instana within a time window, given by the from_time
+    and the to_time
+    """
+
+    if from_time:
```
Comment (Lawouach) [inline chaosinstana/probes.py]: With the change above, you can now remove the if.

File: chaosinstana/probes.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        from_time = convert_time(from_time)
+
+    if to_time:
```
Comment (Lawouach) [inline chaosinstana/probes.py]: With the change above, you can now remove the if.

**Review by Lawouach — CHANGES_REQUESTED** [review]
LbetterTM :)

A few changes requested.
Also you are missing the changelog entry.

**Comment by Lawouach** [comment]
Why all these conflicts? :)

**Comment by gtfisher** [comment]
@Lawouach I think we're ready now to merge barring any final changes?

File: chaosinstana/api.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            configuration.get("instana_host"))
+
+    params = {}
+    if from_time:
+        params["from"] = from_time
+
+    if to_time:
+        params["to"] = to_time
+
+    logger.debug("url: {}".format(url))
```
Comment (Lawouach) [inline chaosinstana/api.py:35]: More a comment than anything else, but those debug statement should usually remain only if they help investigation later on :)

File: chaosinstana/probes.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+def get_all_events_in_window(from_time: str = None, to_time: str = None,
+                             configuration: Configuration = None,
+                             secrets: Secrets = None) -> str:
+    """
+    get all events from instana within a time window, given by the from_time
+    and the to_time
+    """
+    from_time = convert_time(from_time)
+    to_time = convert_time(to_time)
+    logger.debug("from_time {}".format(from_time))
```
Comment (Lawouach) [inline chaosinstana/probes.py:21]: same here :)

**Review by Lawouach — APPROVED** [review]
LGTM

**Comment by Lawouach** [comment]
Thanks for your hardwork!

You can happily merge :)

**Pull request merged**